### PR TITLE
Make mobile work, add dark mode, add defaultOpen prop to overlay

### PR DESCRIPTION
**Pull request by vanillacode314**

### CODE REVIEW

**Review by thetarnav — APPROVED** [review]
Ok wow, this is sick!
Thanks for contributing this :)
So nice to see it on mobile, and the improvements to smaller screens should also apply to the extension, as you can have it open in a pretty slim window too.

I have some minor issues with some parts of the mobile experience, but the PR is fine as is, you don't have to address them if you don't want to. Probably just something to do in the future:
- You cannot really resize the overlay nor the inspector panel as you could on a desktop. The dragging just isn't working, it stops right after you try to move.
- The tree view isn't recalculating the visible rows if the container height changes. I didn't do that previously because the container height was almost always the same, but now it's more visible than before. You can see it when you open the inspector, scroll the tree view to recalculate, and close the inspector.
- If you click on a tree node, it becomes selected immediately, but it is also in the hovered state—so if a component gets selected its element will be highlighted on the page until you click somewhere else. Not sure if it's that bad though, it's not like you have any other way to hover things on mobile. But there needs to be a decision made for mobile: do we highlight the element always, or never?
- The "X" button of the side-panel probably should be bigger on mobile. You can barely tap it. This probably also applies to the button toggling the overlay, it's better but still a bit small.

File: packages/overlay/src/controller.ts [hunk 1]
```diff
@@ -7,9 +7,7 @@ import { Messages } from '@solid-devtools/shared/bridge'
 
 const { setEnabled, enabled, debuggerData } = createInternalRoot(() => {
   const [userEnabled, setEnabled] = createSignal(false)
-  // the devtools overlay isn't supporting mobile devices at the moment
-  const isMobile = createMediaQuery('(max-width: 640px)')
```
Comment (thetarnav) [inline packages/overlay/src/controller.ts:11]: The import of `createMediaQuery` should be removed too

File: packages/overlay/src/index.tsx [hunk 2]
```diff
@@ -28,8 +34,10 @@ export const DevtoolsOverlay: Component = props => {
 }
 
 const Overlay: Component = props => {
+  const isMobile = createMediaQuery('(max-width: 640px)')
   const [progress, setProgress] = createSignal(0.5)
   const [dragging, setDragging] = createSignal(false)
+  createRenderEffect(() => setProgress(isMobile() ? 0.8 : 0.5))
```
[the remark this answers is not in the document]
Comment (vanillacode314) [reply packages/overlay/src/index.tsx:40]: honestly, I don't even know the difference XD

File: packages/overlay/src/index.tsx [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 import overlayStyles from './styles.css'
 
-export const DevtoolsOverlay: Component = props => {
+interface Props {
+  defaultOpen?: boolean
+}
+
+export const DevtoolsOverlay: Component<Props> = props => {
   let dispose: VoidFunction | undefined
+  onMount(() => props.defaultOpen && setEnabled(true))
```
[the remark this answers is not in the document]
Comment (vanillacode314) [reply packages/overlay/src/index.tsx:20]: yes debugger data is blank if it's set immediately

**Comment by vanillacode314** [comment]
> * You cannot really resize the overlay nor the inspector panel as you could on a desktop. The dragging just isn't working, it stops right after you try to move.

Mobile has pull to reload and other weird stuff with touch. I believe it would be best to not have resizing option on mobile

**Comment by vanillacode314** [comment]
> * If you click on a tree node, it becomes selected immediately, but it is also in the hovered state—so if a component gets selected its element will be highlighted on the page until you click somewhere else. Not sure if it's that bad though, it's not like you have any other way to hover things on mobile. But there needs to be a decision made for mobile: do we highlight the element always, or never?

probably doesn't matter as most of the time someone would have the mobile device connected to a desktop while debugging or use the browser emulation which would probably but if we had to do something I would say add a small toggle besides node names that toggles the boundary

**Comment by vanillacode314** [comment]
> * The tree view isn't recalculating the visible rows if the container height changes. I didn't do that previously because the container height was almost always the same, but now it's more visible than before. You can see it when you open the inspector, scroll the tree view to recalculate, and close the inspector.

I can look into this if you could redirect me to some relevant files I should look into

File: packages/overlay/src/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (thetarnav) [reply packages/overlay/src/index.tsx:20]: hmmm
guess I'll have to investigate later
it's fine as is then

**Comment by thetarnav** [comment]
> Mobile has pull to reload and other weird stuff with touch. I believe it would be best to not have resizing option on mobile
> Probably display hidden on drag div on mobile?

Yup, that's a good solution for me. It's not needed there anyway. But it is needed for resizing side-panel on the extension where you can use the mouse. As the mobile version is detected just by the screen width, this will also influence the extension. I think there is a way to target devices with touch as the preferred input.

As the inspector cannot be resized, shouldn't it cover more of the screen? Like how the overlay is covering 80% instead.



> > * If you click on a tree node, it becomes selected immediately, but it is also in the hovered state—so if a component gets selected its element will be highlighted on the page until you click somewhere else. Not sure if it's that bad though, it's not like you have any other way to hover things on mobile. But there needs to be a decision made for mobile: do we highlight the element always, or never?
> 
> probably doesn't matter as most of the time someone would have the mobile device connected to a desktop while debugging or use the browser emulation which would probably but if we had to do something I would say add a small toggle besides node names that toggles the boundary

Good idea. Having it togglable with a button somewhere on top of the inspector would be nice.




> > * The tree view isn't recalculating the visible rows if the container height changes. I didn't do that previously because the container height was almost always the same, but now it's more visible than before. You can see it when you open the inspector, scroll the tree view to recalculate, and close the inspector.
> 
> I can look into this if you could redirect me to some relevant files I should look into

Here is where the virtualisation happens: https://github.com/thetarnav/solid-devtools/blob/main/packages/frontend/src/modules/structure/Structure.tsx

**Comment by thetarnav** [comment]
Another thing that I just noticed is that the scrolling containers on the overlay should have `overscroll-behavior` disabled, to not mess with the page scroll if reached the end

**Comment by vanillacode314** [comment]
> Another thing that I just noticed is that the scrolling containers on the overlay should have `overscroll-behavior` disabled, to not mess with the page scroll if reached the end

Also I see you are using overflow: overlay which is not standard and doesn't work in all browsers

**Comment by vanillacode314** [comment]
> As the inspector cannot be resized, shouldn't it cover more of the screen? Like how the overlay is covering 80% instead.

50/50 ?

**Comment by thetarnav** [comment]
> Also I see you are using overflow: overlay which is not standard and doesn't work in all browsers

ah that's true
previously this was only a chrome extension, so I had the luxury of not caring about that :D

> 50/50 ?

yup, something like that, thank you

**Comment by thetarnav** [comment]
doing that now :)
just wanted to make sure everything builds, and make some changes here and there

**Pull request merged**